lib/house.rb

```diff
@@ -0,0 +1,54 @@
+class House
+
+  attr_reader :address, :price, :rooms
+
+  def initialize(price, address)
+    @price = price[1..-1].to_i
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    rooms << room
+  end
+
+  def above_market_average?
+    price > 500000
+  end
+
+  def rooms_from_category(category)
+    rooms.find_all { |room| room.category == category }
+  end
+
+  def area
+    total_area = []
+    rooms.each { |room| total_area << room.area }
+    total_area.sum
```

Comment:
This works fine, but remember-- sum is an enumerable as well, so you're iterating over the whole collection twice (not a big deal when you only have 4 rooms, but if you have 4 thousand or 4 million, it would be a real big deal).  So instead of iterating twice here, you could just do:

```
def area 
       rooms.sum { |room| total_area << room.area }
end
```